specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,668 @@
+These attributes are abstracted with the TextBox component APIs:
+
+-   TextBox(String id)
+-   SetPlaceHolderContent(String placeHolderContent)
+
+Below is an example use:
+
+-   AppNotificationContent
+
+```c#
+AppNotificationContent()
+    .AddTextBox(TextBox(L"textBox")
+        .SetPlaceHolderContent(L"reply"))
+    .AddButton(Button(L"Send").
+        .SetInputId(L"textBox"))
+    .Show();
+```
+
+![TextBox Example](TextBoxExample.png)
+
+**SelectionMenu** SelectionMenu is an <input> component that allows users to pick from a dropdown
+menu of values on an AppNotification.
+
+Here is the schema for input when type="selection" <input id = string defaultInput? = string >
+
+<!-- Child elements --> selection{0,5} </input>
+
+These attributes are abstracted with the SelectionMenu component APIs:
+
+-   SelectionMenu(String id)
+-   SetDefaultSelection(String id)
+-   AddSelection(String id, String content)
+
+Below is an example use:
+
+-   AppNotificationContent
+
+```c#
+AppNotificationContent()
+    .AddSelectionMenu(SelectionMenu(L"selectionMenu")
+        .SetDefaultSelection(L"yes")
+        .AddSelection(L"yes", L"Going")
+        .AddSelection(L"maybe", L"Maybe")
+        .AddSelection(L"no", L"Decline"))
+    .AddButton(Button(L"Send")
+        .SetInputId(L"selectionMenu"))
+    .Show();
+```
+
+![SelectionMenu Example](SelectionMenuExample.png)
+
+# API Details
+
+```c#
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass Text
+    {
+        Text(String content);
+
+        String Content{ get; };
+        Text SetContent(String content);
+
+        // The target locale of the XML payload, specified as a BCP-47 language tags.
+        String Language{ get; };
+        Text SetLanguage(String language);
+
+        // Center the text for incoming call notifications. This value is only used for notifications with with a scenario value of IncomingCall.
+        Boolean IsCallScenario { get; };
+        Text SetIsCallScenario(Boolean isCallScenario);
+
+        // Displays text at the bottom of your notification, along with your app's identity or the notification's timestamp.
+        // On older versions of Windows that don't support attribution text, the text will simply be displayed as another text element.
+        Boolean IsAttributionText { get; };
+        Text SetIsAttributionText(Boolean isAttributionText);
+
+        // Retrieves the XML content of the Text.
+        String GetXml();
+    }
+
+    enum ButtonStyle
+    {
+        Success, // The button will be green
+        Critical, // The button will be red
+    }
+
+    runtimeclass Button
+    {
+        Button();
+        Button(String content);
+
+        // Get/Sets the text to display on the button.
+        String Content{ get; };
+        Button SetContent(String content);
+
+        // Gets app-defined string of arguments that the app can later retrieve once it is activated when the user clicks the button.
+        String Arguments { get; };
+
```

Comment:
Should we really be return a string here? Shouldn't we be returning something easier to manipulate, like a map since arguments are key / value pairs?

Reply:
Maybe we don't need to expose this. Is there a scenario in which a developer needs to have access to the arguments that have been added to a builder?